HHS/Head-Start-TTADP
# [PROD] Global Try Catch, Objective Order (AR/RTR), Audit Vulns

## Description of change

**TTAHUB-1147:** Maintain ordering of Objectives created via AR
**TTAHUB-1202:** Maintain ordering of Objectives on the RTR
**TTAHUB-1246:** The controls of the topic/frequency graph do not scroll with the rest of the graph
**TTAHUB-1229:** Fixed a bug where it was possible to cause a javascript error when unlocking reports.
**TTAHUB-1240:** Prevent selecting of objectives with leading or trailing white space.

Also updated documentation about recreating an associated service.

**12/02 Fix audit vulnerabilities :**
There are three audit vulnerabilities 2 we can use the patched version (minimatch and decode-uri-component). The third loader-utils doesn't support the patched version. However because its used by react-scripts can be safely ignored.

## How to test

**TTAHUB-1147:** Objectives created via AR should maintain the order they are created in.
**TTAHUB-1202:** Add objectives to the RTR; save, leave, return. They'll be in the same order.
**TTAHUB-1246:** On the regional dashboard, scroll the long bar graph. You'll see that the controls stay in their original positions.
**TTAHUB-1229:** Hard to test - the main way to do it would be unlock a report and check the server output. There will be no errors.
**TTAHUB-1240:** Prevent selecting of objectives with leading or trailing white space.

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1147
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1202
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1246
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1229
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1240

## Checklists

### Every PR



### Production Deploy


### After merge/deploy
